sdks/go/pkg/beam/core/funcx/fn_test.go

```diff
@@ -291,6 +301,96 @@ func TestEmits(t *testing.T) {
 	}
 }
 
+func TestPane(t *testing.T) {
+	tests := []struct {
+		Name   string
+		Params []FnParamKind
+		Pos    int
+		Exists bool
+	}{
+		{
+			Name:   "pane input",
+			Params: []FnParamKind{FnContext, FnPane},
+			Pos:    1,
+			Exists: true,
+		},
+		{
+			Name:   "no pane input",
+			Params: []FnParamKind{FnContext, FnEventTime},
+			Pos:    -1,
+			Exists: false,
+		},
+	}
+
+	for _, test := range tests {
+		test := test
+		t.Run(test.Name, func(t *testing.T) {
+			// Create a Fn with a filled params list.
+			params := make([]FnParam, len(test.Params))
+			for i, kind := range test.Params {
+				params[i].Kind = kind
+				params[i].T = nil
+			}
+			fn := new(Fn)
+			fn.Param = params
```

Comment:
It's in very rare circumstances where one calls "new" for idiomatic go, to get a pointer value.
```suggestion
			fn := &Fn{Params: params}
```

Reply:
I agree.